docs/content/docs/deployment/memory/network_mem_tuning.md

```diff
@@ -0,0 +1,176 @@
+
+#### Multiple inputs and unions
+
+Currently, the throughput calculation and buffer debloating happens on the subtask level. If your subtask has
+multiple different inputs, or it has single but unioned input, this can cause some problems. It might be especially
+visible if those different inputs have vastly different throughputs. In such case as the result of buffer debloating
+low throughput input might have still too much buffered in-flight data, while the hight throughput input might
+have too small buffers to sustain that throughput. We recommend paying special attention to such subtasks
+when testing this feature.
+
+#### Buffer size and number of buffers
+
+Currently, buffer debloating only caps the maximal used buffer size. The actual buffer size and the number of
+buffers remains unchanged. One consequence of this is that the debloating can not reduce memory usage of
+your job. To achieve that, you have to manually reduce either the amount or the size of the buffers. Secondly
+if you want to further reduce the amount of buffered in-flight data below what buffer debloating allows currently,
+you still might want to manually reduce the configured number of buffers.
+
+## Network buffer lifecycle
+Logically, Flink has several local buffer pools one for output stream and one for each input gate. 
+Each of that pools is limited to at most 
+```
+#channels * taskmanager.network.memory.buffers-per-channel + taskmanager.network.memory.floating-buffers-per-gate
+```
+
+The size of the buffer can be configured by setting `taskmanager.memory.segment-size`.
+
+### Input network buffers
+Buffers in the input channel are divided into exclusive and floating buffers.
+Flink attempts to acquire the configured amount of the exclusive buffers in the initialization phase for each channel.
+Exclusive buffers can be used only by one particular channel.
+A channel can request additional floating buffers from a buffer pool shared across all channels belonging to the given input gate.
+All exclusive buffers must be fulfilled on initialization, otherwise job will fail with an exception.
+On top of that single floating buffer has to be allocated during initialization for Flink to be able to make progress.
+Remaining floating buffers are optional and are acquired only if there are enough resource available.
+
+### Output network buffers
+Unlike the input buffer pool, the output buffers pool have only one type of buffers which it shares all among all subpartitions.
+In order to avoid the excessive data skew, the number of buffers for each subpartition is limited by `taskmanager.network.memory.max-buffers-per-channel`.
+
+Similarly, as on the input side, the configured amount of the exclusive buffers and floating buffers is treated only
+as the recommended values. If there are not enough buffers available, Flink will be able to make a progress with
+only a single exclusive buffer per output subpartition and zero floating buffers.
+
```

Comment:
```suggestion
## Network buffer lifecycle
 
Flink has several local buffer pools - one for the output stream and one for each input gate. 
Each of those pools is limited to at most 
```
# channels * taskmanager.network.memory.buffers-per-channel + taskmanager.network.memory.floating-buffers-per-gate
```

The size of the buffer can be configured by setting `taskmanager.memory.segment-size`.

### Input network buffers

Buffers in the input channel are divided into exclusive and floating buffers.  Exclusive buffers can be used by only one particular channel.  A channel can request additional floating buffers from a buffer pool shared across all channels belonging to the given input gate. The remaining floating buffers are optional and are acquired only if there are enough resources available.

In the initialization phase:
- Flink will try to acquire the configured amount of exclusive buffers for each channel
- all exclusive buffers must be fulfilled or the job will fail with an exception
- a single floating buffer has to be allocated for Flink to be able to make progress

### Output network buffers

Unlike the input buffer pool, the output buffer pool has only one type of buffer which it shares among all subpartitions.

In order to avoid excessive data skew, the number of buffers for each subpartition is limited by the `taskmanager.network.memory.max-buffers-per-channel` setting.

Like the input buffer pool, the configured amount of exclusive buffers and floating buffers is only treated as recommended values. If there are not enough buffers available, Flink can make progress with only a single exclusive buffer per output subpartition and zero floating buffers.

```